dotnet/Capstone/Controllers/BirdListController.cs

```diff
@@ -0,0 +1,158 @@
+﻿using Capstone.DAO;
+using Capstone.Exceptions;
+using Capstone.Models;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+
+namespace Capstone.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class BirdListController: ControllerBase
+    {
+        private readonly IBirdListDao birdListDao;
+
+        public BirdListController(IBirdListDao birdListDao)
+        {
+            this.birdListDao = birdListDao;
+        }
+
+
+
+        [HttpGet("/lists/{listId}")]
+        public IActionResult getList(int listId)
+        {
+            const string errorMessage = "No bird matches this id.";
+            IActionResult result;
+
+            try
+            {
+                BirdList birdList = birdListDao.getList(listId);
+
+                if (birdList != null)
+                {
+                    result = Ok(birdList);
+                }
+                else
+                {
+                    result = NotFound(new { message = errorMessage });
+                }
+            }
+            catch (DaoException e)
+            {
+                result = NotFound(new { message = e.Message });
+            }
+
+
+            return result;
+
+        }
+
+
+        [HttpGet("/lists/")]
+        public List<BirdList> getAllLists()
+        {
+            IIdentity user = User.Identity;
+
+            List<BirdList> birdList = birdListDao.getAllLists(user.Name);
+
+            if (birdList == null)
+            {
+                Console.WriteLine("No lists");
+            }
+
+            return birdList;
+        }
+
+
+        [HttpPut("/editList/{listId}")]
+        public IActionResult editList(BirdList list, int listId)
+        {
+            const string errorMessage = "An error occurred and list could not be modified.";
+            
+            IActionResult result;
+
+            try
+            {
+                birdListDao.editList(list, listId);
```

Comment:
Nothing prevents a user from changing another user's list. Only the name is effected by this, so I'm good to save it for the database-rework to main merge